### ISSUE TITLE
install helper binaries error 

### ISSUE THREAD

**Issue by zach007** [begriffs/haskell-vim-now#272]
system : ubuntun 14
stack version : 1.5

Installing helper binaries...
    stack --resolver lts-9.10 install ghc-mod --install-ghc --verbosity warning
    stack --resolver lts-9.10 install cabal-install --install-ghc --verbosity warning
    stack --resolver lts-8.14 install hindent --install-ghc --verbosity warning
name:                dependencies
version:             0.1.0.0
synopsis:            helper binaries for vim
homepage:            https://github.com/begriffs/haskell-vim-now
license:             MIT
author:              Joe Nelson
maintainer:          cred+github@begriffs.com
category:            Development
build-type:          Simple
cabal-version:       >=1.10

library
  build-depends:       base >=4.9 && <4.10
                     , apply-refact
                     , codex
                     , hasktags
                     , hlint
                     , hoogle
                     , hscope
                     , pointfree
                     , pointful
  default-language:    Haskell2010
Looking for .cabal or package.yaml files to use to init the project.
Using cabal packages:
- dependencies.cabal

Unable to download snapshot list, and therefore could not generate a stack.yaml file automatically
This sometimes happens due to missing Certificate Authorities on your system. For more information, see:

    https://github.com/commercialhaskell/stack/issues/234

You can try again, or create your stack.yaml file by hand. See:

    http://docs.haskellstack.org/en/stable/yaml_configuration/

Exception was: HttpExceptionRequest Request {
  host                 = "s3.amazonaws.com"
  port                 = 443
  secure               = True
  requestHeaders       = []
  path                 = "/haddock.stackage.org/snapshots.json"
  queryString          = ""
  method               = "GET"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
 ConnectionTimeout

"stack init --solver --install-ghc" failed with error 1
*** setup_haskell.hs failed with error 1.
*** Aborting...

**Comment by zach007** [comment]
Installing helper binaries...
    stack --resolver lts-9.10 install ghc-mod --install-ghc --verbosity warning
    stack --resolver lts-9.10 install cabal-install --install-ghc --verbosity warning
    stack --resolver lts-8.14 install hindent --install-ghc --verbosity warning
name:                dependencies
version:             0.1.0.0
synopsis:            helper binaries for vim
homepage:            https://github.com/begriffs/haskell-vim-now
license:             MIT
author:              Joe Nelson
maintainer:          cred+github@begriffs.com
category:            Development
build-type:          Simple
cabal-version:       >=1.10

library
  build-depends:       base >=4.9 && <4.10
                     , apply-refact
                     , codex
                     , hasktags
                     , hlint
                     , hoogle
                     , hscope
                     , pointfree
                     , pointful
  default-language:    Haskell2010
Looking for .cabal or package.yaml files to use to init the project.
Using cabal packages:
- dependencies.cabal

Selecting the best among 11 snapshots...

* Partially matches lts-9.10
    apply-refact not found
        - dependencies requires -any
    codex not found
        - dependencies requires -any
    hasktags not found
        - dependencies requires -any
    hscope not found
        - dependencies requires -any
    pointfree not found
        - dependencies requires -any

* Rejected nightly-2017-10-26
    ghc-8.2.1 cannot be used for these packages:
        - dependencies
    base version 4.10.0.0 found
        - dependencies requires >=4.9 && <4.10

* Partially matches lts-8.24
    apply-refact not found
        - dependencies requires -any
    codex not found
        - dependencies requires -any
    hasktags not found
        - dependencies requires -any
    hscope not found
        - dependencies requires -any
    pointfree not found
        - dependencies requires -any

* Partially matches lts-7.24
    codex not found
        - dependencies requires -any
    hasktags not found
        - dependencies requires -any
    hscope not found
        - dependencies requires -any
    pointfree not found
        - dependencies requires -any

* Rejected lts-6.35
    ghc-7.10.3 cannot be used for these packages:
        - dependencies
    base version 4.8.2.0 found
        - dependencies requires >=4.9 && <4.10

* Rejected lts-5.18
    ghc-7.10.3 cannot be used for these packages:
        - dependencies
    base version 4.8.2.0 found
        - dependencies requires >=4.9 && <4.10

* Rejected lts-4.2
    ghc-7.10.3 cannot be used for these packages:
        - dependencies
    base version 4.8.2.0 found
        - dependencies requires >=4.9 && <4.10

* Rejected lts-3.22
    ghc-7.10.2 cannot be used for these packages:
        - dependencies
    base version 4.8.1.0 found
        - dependencies requires >=4.9 && <4.10

* Rejected lts-2.22
    ghc-7.8.4 cannot be used for these packages:
        - dependencies
    base version 4.7.0.2 found
        - dependencies requires >=4.9 && <4.10

* Rejected lts-1.15
    ghc-7.8.4 cannot be used for these packages:
        - dependencies
    base version 4.7.0.2 found
        - dependencies requires >=4.9 && <4.10

* Rejected lts-0.7
    ghc-7.8.3 cannot be used for these packages:
        - dependencies
    base version 4.7.0.1 found
        - dependencies requires >=4.9 && <4.10

Selected resolver: lts-7.24
*** Resolver lts-7.24 will need external packages:
    codex not found
        - dependencies requires -any
    hasktags not found
        - dependencies requires -any
    hscope not found
        - dependencies requires -any
    pointfree not found
        - dependencies requires -any

Using resolver: lts-7.24
Solver requires that cabal be on your PATH
Try running 'stack install cabal-install'

"stack init --solver --install-ghc" failed with error 1
*** setup_haskell.hs failed with error 1.
*** Aborting...

**Comment by begriffs** [comment]
Thanks for reporting, I'll try to reproduce on a clean virtual machine and let you know how it goes.

**Comment by begriffs** [comment]
OK, I ran into the "cabal not on path" error as well. Started from a fresh server on digital ocean:

* Ubuntu 14.04.5 x64 with 2 GB ram
* Stack version 1.5.1, Git revision 600c1f01435a10d127938709556c1682ecfd694e (4861 commits) x86_64 hpack-0.17.1
* Installing with a non-root but sudo user

I was able to get past the error by modifying my PATH:

```bash
export PATH="$HOME/.local/bin:$PATH"
```

That's the directory Stack saves binaries it has built, such as cabal. (I was able to determine this path by running `stack path --local-bin`.)

**Comment by begriffs** [comment]
Ugh, spoke too soon. My installer ended up breaking when trying to install codex.

Perhaps we ought to pin an LTS stackage version for each and every helper binary rather than attempting to use the Stack solver to install a bunch of them at once. This is so touchy it always breaks.

Output of failure:

```
stack --resolver lts-7.24 install codex-0.5.1.2 --install-ghc --verbosity warning
Progress: 75/94
--  While building package lens-4.14 using:
      /tmp/stack22545/lens-4.14/.stack-work/dist/x86_64-linux/Cabal-1.24.0.0/setup/setup --builddir=.stack-work/dist/x86_64-linux/Cabal-1.24.0.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure (-11)
    Logs have been written to: /home/j/.config/haskell-vim-now/hvn-helper-binaries/.stack-work/logs/lens-4.14.log

    [1 of 2] Compiling Main             ( /tmp/stack22545/lens-4.14/Setup.lhs, /tmp/stack22545/lens-4.14/.stack-work/dist/x86_64-linux/Cabal-1.24.0.0/setup/Main.o )

    /tmp/stack22545/lens-4.14/Setup.lhs:55:69: warning: [-Wdeprecations]
        In the use of type constructor or class ‘InstalledPackageId’
        (imported from Distribution.Package):
        Deprecated: "Use UnitId instead"
    [2 of 2] Compiling StackSetupShim   ( /home/j/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs, /tmp/stack22545/lens-4.14/.stack-work/dist/x86_64-linux/Cabal-1.24.0.0/setup/StackSetupShim.o )
    Linking /tmp/stack22545/lens-4.14/.stack-work/dist/x86_64-linux/Cabal-1.24.0.0/setup/setup ...
    Configuring lens-4.14...
    Building lens-4.14...
    Preprocessing library lens-4.14...
    [ 1 of 81] Compiling Paths_lens       ( .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/autogen/Paths_lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Paths_lens.o )
    [ 2 of 81] Compiling Data.Map.Lens    ( src/Data/Map/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/Map/Lens.o )
    [ 3 of 81] Compiling Control.Lens.Internal.List ( src/Control/Lens/Internal/List.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/List.o )
    [ 4 of 81] Compiling Control.Lens.Internal.TH ( src/Control/Lens/Internal/TH.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/TH.o )
    [ 5 of 81] Compiling Control.Lens.Internal.Coerce ( src/Control/Lens/Internal/Coerce.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Coerce.o )
    [ 6 of 81] Compiling Control.Lens.Internal.Instances ( src/Control/Lens/Internal/Instances.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Instances.o )
    [ 7 of 81] Compiling Control.Lens.Internal.Zoom ( src/Control/Lens/Internal/Zoom.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Zoom.o )
    [ 8 of 81] Compiling Control.Lens.Internal.Setter ( src/Control/Lens/Internal/Setter.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Setter.o )
    [ 9 of 81] Compiling Control.Lens.Internal.Review ( src/Control/Lens/Internal/Review.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Review.o )
    [10 of 81] Compiling Control.Lens.Internal.Prism ( src/Control/Lens/Internal/Prism.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Prism.o )
    [11 of 81] Compiling Control.Lens.Internal.Iso ( src/Control/Lens/Internal/Iso.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Iso.o )
    [12 of 81] Compiling Control.Lens.Internal.Level ( src/Control/Lens/Internal/Level.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Level.o )
    [13 of 81] Compiling Control.Lens.Internal.Indexed ( src/Control/Lens/Internal/Indexed.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Indexed.o )
    [14 of 81] Compiling Control.Lens.Internal.Context ( src/Control/Lens/Internal/Context.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Context.o )
    [15 of 81] Compiling Control.Lens.Internal.Bazaar ( src/Control/Lens/Internal/Bazaar.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Bazaar.o )
    [16 of 81] Compiling Control.Lens.Internal.Magma ( src/Control/Lens/Internal/Magma.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Magma.o )
    [17 of 81] Compiling Control.Lens.Internal.Getter ( src/Control/Lens/Internal/Getter.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Getter.o )
    [18 of 81] Compiling Control.Lens.Internal.Fold ( src/Control/Lens/Internal/Fold.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Fold.o )
    [19 of 81] Compiling Control.Lens.Type ( src/Control/Lens/Type.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Type.o )
    [20 of 81] Compiling Control.Lens.Setter ( src/Control/Lens/Setter.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Setter.o )
    [21 of 81] Compiling Control.Lens.Lens ( src/Control/Lens/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Lens.o )
    [22 of 81] Compiling Control.Lens.Tuple ( src/Control/Lens/Tuple.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Tuple.o )
    [23 of 81] Compiling Control.Lens.Getter ( src/Control/Lens/Getter.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Getter.o )
    [24 of 81] Compiling Control.Lens.Review ( src/Control/Lens/Review.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Review.o )
    [25 of 81] Compiling Control.Lens.Prism ( src/Control/Lens/Prism.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Prism.o )
    [26 of 81] Compiling Control.Lens.Zoom ( src/Control/Lens/Zoom.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Zoom.o )
    [27 of 81] Compiling Data.Set.Lens    ( src/Data/Set/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/Set/Lens.o )
    [28 of 81] Compiling Data.HashSet.Lens ( src/Data/HashSet/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/HashSet/Lens.o )
    [29 of 81] Compiling Control.Lens.Fold ( src/Control/Lens/Fold.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Fold.o )
    [30 of 81] Compiling Control.Lens.Traversal ( src/Control/Lens/Traversal.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Traversal.o )
    [31 of 81] Compiling Control.Lens.Indexed ( src/Control/Lens/Indexed.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Indexed.o )
    [32 of 81] Compiling Control.Lens.Level ( src/Control/Lens/Level.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Level.o )
    [33 of 81] Compiling Control.Lens.Reified ( src/Control/Lens/Reified.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Reified.o )
    [34 of 81] Compiling Data.Data.Lens   ( src/Data/Data/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/Data/Lens.o )
    [35 of 81] Compiling Control.Lens.Plated ( src/Control/Lens/Plated.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Plated.o )
    [36 of 81] Compiling Control.Lens.Internal.ByteString ( src/Control/Lens/Internal/ByteString.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/ByteString.o )
    [37 of 81] Compiling Control.Lens.Equality ( src/Control/Lens/Equality.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Equality.o )
    [38 of 81] Compiling Control.Lens.Iso ( src/Control/Lens/Iso.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Iso.o )
    [39 of 81] Compiling Control.Lens.Wrapped ( src/Control/Lens/Wrapped.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Wrapped.o )
    [40 of 81] Compiling Data.Vector.Generic.Lens ( src/Data/Vector/Generic/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/Vector/Generic/Lens.o )
    [41 of 81] Compiling Data.Text.Strict.Lens ( src/Data/Text/Strict/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/Text/Strict/Lens.o )
    [42 of 81] Compiling Data.Text.Lazy.Lens ( src/Data/Text/Lazy/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/Text/Lazy/Lens.o )
    [43 of 81] Compiling Data.Text.Lens   ( src/Data/Text/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Data/Text/Lens.o )
    [44 of 81] Compiling Control.Lens.Empty ( src/Control/Lens/Empty.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Empty.o )
    [45 of 81] Compiling Control.Lens.Each ( src/Control/Lens/Each.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Each.o )
    [46 of 81] Compiling Control.Lens.Cons ( src/Control/Lens/Cons.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Cons.o )
    [47 of 81] Compiling Control.Lens.Internal.Deque ( src/Control/Lens/Internal/Deque.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/Internal/Deque.o )
    [48 of 81] Compiling Control.Lens.At  ( src/Control/Lens/At.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.0.0/build/Control/Lens/At.o )
"stack --resolver lts-7.24 install codex-0.5.1.2 --install-ghc --verbosity warning" failed with error 1
*** setup_haskell.hs failed with error 1.
*** Aborting...
```

**Comment by zach007** [comment]
thanks!

**Comment by jship** [comment]
Related to #270.

**Comment by jship** [comment]
Welp, the plot thickens.  I successfully installed haskell-vim-now on a clean Ubuntu VM: [log as gist](https://gist.github.com/jship/855ae7df1d2c37d74adbe63706e9b354)

I did not have to do any path finagling and the whole thing was a success.  No issue with [installing codex](https://gist.github.com/jship/855ae7df1d2c37d74adbe63706e9b354#file-hvn_20171108-txt-L264) either.

* Ubuntu 16.04.2, 6GB RAM
* Stack version 1.5.1, Git revision 600c1f01435a10d127938709556c1682ecfd694e (4861 commits) x86_64 hpack-0.17.1

As far as cabal 1 vs cabal 2, [LTS-9.12](https://www.stackage.org/lts-9.12/package/cabal-install-1.24.0.2) has cabal cabal-install-1.24.0.2.  LTS-9.12 is what my VM got in the global stack config from a fresh install of stack.  cabal-install-1.24.0.2 is the same version @zach007 got from LTS-9.10.

This bit is strange to me from @zach007's log:

```bash
Solver requires that cabal be on your PATH
Try running 'stack install t'
```

Where is the `t` coming from?
